lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -2828,6 +2867,16 @@ instance EncodeAddress n => ToJSON (ApiTxInput n) where
         fromValue (Object o) = o
         fromValue _ = mempty
 
+instance DecodeAddress n => FromJSON (ApiTxCollateral n) where
+    parseJSON v = ApiTxCollateral <$> optional (parseJSON v) <*> parseJSON v
+
+instance EncodeAddress n => ToJSON (ApiTxCollateral n) where
+    toJSON (ApiTxCollateral s i) =
+        Object (maybe mempty (fromValue . toJSON) s <> fromValue (toJSON i))
+      where
+        fromValue (Object o) = o
+        fromValue _ = mempty
```

Comment:
These instances are basically copies of the `ApiTxInput` instances. Perhaps we can simplify them, and use a more generic deriving style.